tutorials/asr/04_Online_Offline_Speech_Commands_Demo.ipynb

```diff
@@ -0,0 +1,732 @@
+    "!pip install unidecode\n",
+    "!pip install pyaudio\n",
+    "\n",
+    "# ## Install NeMo\n",
+    "BRANCH = 'v1.0.0b2'\n",
+    "!python -m pip install git+https://github.com/NVIDIA/NeMo.git@$BRANCH#egg=nemo_toolkit[asr]\n",
+    "\n",
+    "## Install TorchAudio\n",
+    "!pip install torchaudio>=0.6.0 -f https://download.pytorch.org/whl/torch_stable.html"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "This notebook demonstrates offline and online (from a microphone's stream in NeMo) speech commands recognition \n",
+    "\n",
+    "It is **not a recommended** way to do inference in production workflows. If you are interested in \n",
+    "production-level inference using NeMo ASR models, please sign-up to Jarvis early access program: https://developer.nvidia.com/nvidia-jarvis"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The notebook requires PyAudio library to get a signal from an audio device.\n",
+    "For Ubuntu, please run the following commands to install it:\n",
+    "```\n",
+    "sudo apt-get install -y portaudio19-dev\n",
+    "pip install pyaudio\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This notebook requires the `torchaudio` library to be installed for MatchboxNet. Please follow the instructions available at the [torchaudio Github page](https://github.com/pytorch/audio#installation) to install the appropriate version of torchaudio.\n",
+    "\n",
+    "If you would like to install the latest version, please run the following command to install it:\n",
+    "\n",
+    "```\n",
+    "conda install -c pytorch torchaudio\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import pyaudio as pa\n",
+    "import os, time\n",
+    "import librosa\n",
+    "import IPython.display as ipd\n",
+    "import matplotlib.pyplot as plt\n",
+    "%matplotlib inline\n",
+    "\n",
+    "import nemo\n",
+    "import nemo.collections.asr as nemo_asr"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# sample rate, Hz\n",
+    "SAMPLE_RATE = 16000"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Restore the model from NGC"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "mbn_model = nemo_asr.models.EncDecClassificationModel.from_pretrained(\"MatchboxNet-3x1x64-v2\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Since speech commands model MatchBoxNet doesn't consider non-speech scenario, \n",
+    "here we use a vad model to help reduce false alarm for background noise/silence.\n",
```

Comment:
Full form of VAD + describe in detail what VAD will perform - background/foreground detection in order to activate MBN inference pipeline